internal/services/http_server/service_test.go

```diff
@@ -0,0 +1,326 @@
+	"github.com/valyala/fasthttp"
+	"github.com/valyala/fasthttp/fasthttputil"
+)
+
+type handlerStub struct{}
+
+func (s handlerStub) Handle(_ *fasthttp.RequestCtx) {}
+
+func TestServiceStartStop(t *testing.T) {
+	h := &handlers.Handlers{Metrics: &handlerStub{}, VkCallbackHandler: &handlerStub{}}
+	s := New(&Config{Address: "localhost", Port: 1337}, h, slog.Default())
+
+	var ok bool
+	p := gomonkey.ApplyFunc(net.Listen, func(network string, address string) (net.Listener, error) {
+		if ok {
+			return fasthttputil.NewInmemoryListener(), nil
+		}
+		return nil, fmt.Errorf("error")
+	})
+	defer p.Reset()
+
+	ok = false
+	err := s.Run(context.Background())
+	assert.EqualError(t, err, "error")
+
+	ok = true
+	errCh := make(chan error)
+	ctx, cancel := context.WithCancel(context.Background())
+	go func() {
+		errCh <- s.Run(ctx)
+	}()
+	cancel()
+	assert.NoError(t, <-errCh)
+}
+
+func TestVkHandler(t *testing.T) {
+	type args struct {
+		hookId             string
+		confirmationString string
+		reqBody            []byte
+		url                string
+	}
+	tests := []struct {
+		name          string
+		args          args
+		statusCode    int
+		bodyContains  []string
+		queueContents []entities.Message
+	}{
+		// general:
+		{
+			name: "json unmarshal error",
+			args: args{
+				reqBody: []byte(`{"type":"",}`),
+				url:     "http://localhost:8080/api/vk/callback/hookId",
+			},
+			statusCode:   400,
+			bodyContains: []string{"json unmarshal error"},
+		},
+		{
+			name: "unsupported message type",
+			args: args{
+				reqBody: []byte(`{"type":"unsupported_message_type","event":"test_event_id","v":"1.0","group_id":12345}`),
+				url:     "http://localhost:8080/api/vk/callback/hookId",
+			},
+			statusCode:   400,
+			bodyContains: []string{"unsupported message type: unsupported_message_type"},
+		},
+		// handleChallenge:
+		{
+			name: "handleChallenge: no hook id",
+			args: args{
+				reqBody: []byte(`{"type":"confirmation","event":"test_event_id","v":"1.0","group_id":12345}`),
+				url:     "http://localhost:8080/api/vk/callback",
+			},
+			statusCode:   404,
+			bodyContains: []string{"Not Found"},
+		},
+		{
+			name: "handleChallenge: interaction not found",
+			args: args{
+				hookId:  "hookId",
+				reqBody: []byte(`{"type":"confirmation","event":"test_event_id","v":"1.0","group_id":12345}`),
+				url:     "http://localhost:8080/api/vk/callback/hookId2",
+			},
+			statusCode:   400,
+			bodyContains: []string{"interaction not found"},
+		},
+		{
+			name: "handleChallenge: ok",
+			args: args{
+				hookId:             "hookId",
+				confirmationString: "confirmationString",
+				reqBody:            []byte(`{"type":"confirmation","event":"test_event_id","v":"1.0","group_id":12345}`),
+				url:                "http://localhost:8080/api/vk/callback/hookId",
+			},
+			statusCode:   200,
+			bodyContains: []string{"confirmationString"},
+		},
+		// handleMessage:
+		{
+			name: "handleChallenge[message_new]: no hook id",
```

Comment:
Here and in all tests below
```suggestion
			name: "handleMessage[message_new]: no hook id",
```